src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -3339,8 +3339,11 @@ internal static string GetDriveQualifiedPath(string path, PSDriveInfo drive)
                 }
                 else
                 {
-                    if (path.StartsWith(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) || 
-                        path.StartsWith(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)) 
+                    // Check if the path begins with "\" or "/" (UNC Path or Path in Unix).
+                    // Ignore if the path resolves to a drive path, this will happen when path is equal to "\\" or "/".
```

Comment:
> equal to "\\" or "/"

One more minor thing. This comment seems not right since the code suggesting `"\" or "/"`.